vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -799,6 +809,24 @@ def get_finished(self) -> tuple[set[str], set[str]]:
                 "and %s requests done recving", self.tp_rank,
                 len(done_sending), len(done_recving))
 
+        # Handle timeout to avoid stranding blocks on remote.
+        now = time.monotonic()
+        timed_out_requests: list[str] = []
+        for req_id, finish_time in self._reqs_to_send.items():
+            if finish_time < 0:
+                # Request just finished, start timeout.
+                self._reqs_to_send[req_id] = now
+            elif now - finish_time >= envs.VLLM_NIXL_ABORT_REQUEST_TIMEOUT:
+                # Timeout exceed, clear the request blocks.
+                timed_out_requests.append(req_id)
+
+        for req_id in timed_out_requests:
+            # Skip communication with other ranks, but
+            if self.tp_rank == 0:
+                self._done_sending_count[req_id] += self.world_size
+                done_sending.add(req_id)
+            del self._reqs_to_send[req_id]
```

Comment:
I think it's better to keep things simple and omit the TP optimization. I think we'll likely make this logic generic and move it outside of the connector impl anyhow (aggregating the finished events in TP case).

Dicts are ordered so we only need to peek the oldest entry.

```suggestion
        # Handle timeout to avoid stranding blocks on remote.
        now = time.time()
        while self._reqs_to_send:
            req_id, expires = next(iter(self._reqs_to_send.items()))
            if now < expires:
                break
            del self._reqs_to_send[req_id]
            done_sending.add(req_id)
```